HACKING.adoc

```diff
@@ -229,6 +229,74 @@ opam switch create my-switch-name --empty
 opam pin add ocaml-variants.$VERSION+branch git+https://$REPO#branch
 ----
 
+=== Incremental builds with `opam`
+
+If you want to fix a bug that needs opam packages to reproduce, or want to try
+a new feature that you're still actively developing on opam packages, you will
+likely need a faster feedback loop than re-creating a new switch or recompiling
+every package each time you make a small change to your code.
+
+There are two mostly orthogonal steps you can take to accelerate this:
+- Use your locally built copy, and only run the install step
+- Prevent your changes from automatically triggering the rebuild of all packages
+
+For the first step, one solution is to use the `--assume-built` flag to
+`opam install` to skip the compilation and only run the installation instructions.
+This, however, requires that your working directory is in a state suitable for
+running `make install`. This means it should be configured to the correct prefix,
+have all the programs that need to be installed built (including `ocamldoc`,
+the `otherlibs` libraries, and `tools`, if you want to install them), and
+you need to be sure that there aren't any mismatch between bytecode and native
+versions of the binaries. If you don't want to rebuild the native versions,
+you need to ensure they get removed before running the install steps.
+
+Here are some sample instructions using the setup from above:
+
+-----
+-opam switch create . --empty
```

Comment:
Unless I missunderstand you need to pin your repo. So either remove that step (and forward to the above full set of preliminary instructions) or repeat all of the instructions.

Reply:
I think the `opam install .` below pins the repo. I can check with opam experts though.